src/Model/Term.php

```diff
@@ -290,4 +290,20 @@ public static function populateTagsFromItem(&$item)
 
 		return $return;
 	}
+
+	/**
+	 * Delete all tags from an item
+	 * @param int itemid - choose from which item the tags will be removed
+	 */
+	public static function deleteAllTags($itemid)
```

Comment:
Please rename to `deleteByItemId` and add a second array parameter `$type` that will be by default `[TERM_HASHTAG, TERM_MENTION]`.

Then please call the method in `Term::insertFromTagFieldByItemId` instead of the first `DBA::delete`.